rest-api/src/main/java/org/jboss/pnc/rest/api/endpoints/SlsaProvenanceV1Endpoint.java

```diff
@@ -0,0 +1,151 @@
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.jboss.pnc.api.slsa.dto.provenance.v1.Provenance;
+import org.jboss.pnc.dto.response.ErrorResponse;
+import org.jboss.pnc.processor.annotation.Client;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+import static org.jboss.pnc.rest.configuration.SwaggerConstants.NOT_FOUND_CODE;
+import static org.jboss.pnc.rest.configuration.SwaggerConstants.NOT_FOUND_DESCRIPTION;
+import static org.jboss.pnc.rest.configuration.SwaggerConstants.SERVER_ERROR_CODE;
+import static org.jboss.pnc.rest.configuration.SwaggerConstants.SERVER_ERROR_DESCRIPTION;
+import static org.jboss.pnc.rest.configuration.SwaggerConstants.SUCCESS_CODE;
+import static org.jboss.pnc.rest.configuration.SwaggerConstants.SUCCESS_DESCRIPTION;
+
+@Tag(name = "Slsa Provenance")
+@Path("/slsa/build-provenance/v1")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Client
+public interface SlsaProvenanceV1Endpoint {
+
+    static final String B_ID = "ID of the build";
+    static final String A_ID = "ID of the artifact";
+    static final String A_PURL = "Purl of the artifact";
+    static final String A_DIGEST = "Digest of the artifact";
+
+    static final String GET_BY_BUILD_ID_DESCR = "Generate the build provenance for a build identified by id.";
+    static final String GET_BY_ARTIFACT_ID_DESCR = "Generate the build provenance for an artifact identified by id.";
+    static final String GET_BY_ARTIFACT_DIGEST_DESCR = "Generate build provenance for an artifact identified by an algorithm-qualified digest (for example: sha256:0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef).";
+    static final String GET_BY_ARTIFACT_PURL_DESCR = "Generate the build provenance for an artifact identified by purl.";
+
+    /**
+     * {@value GET_BY_BUILD_ID_DESCR}
+     *
+     * @param id {@value B_ID}
+     * @return
+     */
+    @Operation(
+            summary = GET_BY_BUILD_ID_DESCR,
+            responses = {
+                    @ApiResponse(
+                            responseCode = SUCCESS_CODE,
+                            description = SUCCESS_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = Provenance.class))),
+                    @ApiResponse(responseCode = NOT_FOUND_CODE, description = NOT_FOUND_DESCRIPTION),
+                    @ApiResponse(
+                            responseCode = SERVER_ERROR_CODE,
+                            description = SERVER_ERROR_DESCRIPTION,
+                            content = @Content(schema = @Schema(implementation = ErrorResponse.class))) })
+    @GET
+    @Path("/build/id/{id}")
```

Comment:
I think `build` should be in plural AFAIR REST recommendations. Like `/builds/id/{id}`.

I am also inclined to make this `/builds/by-id/{id}` but I don't mind it being like it is now.